[qbus] - Update logo / Small code changes

QbusKoen opened the pull request:
Updated logo.

I've updated the logo as mentioned by @Confectrician in https://github.com/openhab/openhab-addons/pull/9191

Signed off by: Koen Schockaert <ks@qbus.be>

QbusKoen commented:
Please advice,

I tried fixing the Qbus binding, which is working fine now.
After my project was fixed, i returnded to this page. See DCO Sccuesful.
Binding runs with Visual Studio Code, think I'm ready to start development on that platform.

Don't know why every developper is listed here!?

Guess i made a mistake somewhere. Can this be fixed?

Confectrician commented:
I think you used an outdated fork of this repository which introduced many differences to the current repositoy contents.
(Like other pull request that have been merged, since the last time you updated your fork.)

But the logo would need to be added in the openhab-docs repository anyway to show up in the official documentation.
I have linked an example in your binding pull request that shows where you have to add a logo to get in into the openHAB website addons page.

Usually there is no log here in the addons repository.
We store them in the docs and the addon page will then use the logo through the generated website.

So the most useful steps would be:

- remove the logo and its reference in your readme completely  in this repository
- add the logo on the openhab/openhab-docs repository
- let the website build do the rest for you without the need to worry about anything further.

If there is still anything to clarify just ping me here and i will try to help. 🙂

QbusKoen commented:
Thank you for pointing me in the right direction, @Confectrician 

**remove the logo and its reference in your readme completely in this repository**
Done in my local copy, that's why i was trying to update this here. Should I close this request and open a new one, to get the logo updated? I must say from the start of this binding i made a mistake in my name for the github repo. Now I changed it from QbusKoen to my real name. Then I pushed the binding, with the updated logo and a small bug fix.
Thern I tried to get the DCO checks all cleared up. So I executed the propesed commands. I had a lot of errors then - seen a lot of my old code - and did the necessary to make everythin good again.
Furthermore it seems to fixed the problem i had (https://community.openhab.org/t/qbus-initial-contribution-unable-to-resolve-root-missing-requirement-root/120676).
Because I prefer working in Visual Studio Code, I decided to move there from Eclipse. After the above steps, i got it working as a charm :-)

**add the logo on the openhab/openhab-docs repository**
-> Done, Thx i was not aware of this. Thoes this mean that my binding will be visible in the addons list after the site is rebuilded?

**let the website build do the rest for you without the need to worry about anything further.**
So i guess i'm all good now? Close this request and make a new one, to get the logo and the bug fix updated?

Many thanks for your advice!
Kind regards,
Koen Schockaert

QbusKoen commented:
@Confectrician, @fwolter
As you can see, i made some little changed to the code. Added the qbus.JPG logo and pushed the new release this morning. Just finished building without errors.

I want to appologize for my ignorance, but i'm keen on learning more.

I've changed my development environment from Eclipse to VS-Code.

Yesterday I made a pull request in the docs for the logo (https://github.com/openhab/openhab-docs/pull/1537)
There i se a markdown error, so i suppose something went wrong.

On my Windows machine I installed the 3 release versions. My binding shows up in the 3.1.0 SNAPSHOT release :-)

I ran the command 'git rebase upstream/main' multiple times, but everytime i ran into troubles. I use GithubDesktop to push my releases. There I see i'm on the main branch.

After several attempts i gave up. Each time i had to accept changes i made in the initial contribution. Maybe this is because i've changed my name for Github... I just don't know.

Some questions:
Can this merge be accepted so my updated code comes on line?
Do I have to wait untill this merge is closed and then try to add the logo again?
How do i get my binding on https://www.openhab.org/addons/ - Just by uploading the logo to the docs?
Now people who want to use this binding have to install the 3.1.0-SNAPSHOT release to find my binding in the addons list. What should I do to get my binding in the stable release?

Again, appologies for my ignorance.
Kind regards,
Koen Schockaert

Confectrician reviewed (commented):
May work for the logo part.
We end up with two logos probably, when the docs repo change is merged.

But we still can remove the logo here afterwards, if it is doubled.

Confectrician commented:
No need to apologize @QbusKoen  🙂 

There are many repositories here and no one can have a complete overview right from the beginning.

fwolter commented on bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java:
```diff
@@ -121,11 +121,25 @@ public synchronized void startCommunication() throws IOException, InterruptedExc
+                qSocket = socket;
+                qOut = new PrintWriter(socket.getOutputStream(), true);
+                qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            } catch (IOException e) {
+
+            } finally {
+                if (socket == null) {
+                    Integer serverCheck = handler.getServerCheck();
+                    handler.bridgeOffline(ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No communication with Qbus Server, will try to reconnect every " + serverCheck
+                                    + " minutes");
+                    return;
```
Are you sure this code is working? The code in `finally` will also be executed if no exception raised.

Please handle the exception appropriately. E.g. set the Thing offline.

fwolter reviewed (changes requested):
Please update the title of this PR, so that one sees that it contains also a code change.

fwolter commented:
> Now people who want to use this binding have to install the 3.1.0-SNAPSHOT release to find my binding in the addons list. What should I do to get my binding in the stable release?

Your binding will be released with the next OH release in May. You don't need to do anything.

QbusKoen commented:
@fwolter thanks for noticing :-)
I've updated the code as you requested, as usually you were right ;-)
In the same commit i also made some changes to improve the readme.md and changed a link.
The command 'git rebase upstream/main' troubles me. I just can't get it. Is this neccesarry for good practice? Or could i just leave it as it is?

fwolter commented on bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java:
```diff
@@ -121,26 +121,29 @@ public synchronized void startCommunication() throws IOException, InterruptedExc
+            qSocket = socket;
+            qOut = new PrintWriter(socket.getOutputStream(), true);
+            qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+        } catch (IOException e) {
+            String msg = e.getMessage();
+            logger.warn("Could not start listening thread, IOException: {}", msg);
+            Integer serverCheck = handler.getServerCheck();
+            handler.bridgeOffline(ThingStatusDetail.COMMUNICATION_ERROR,
+                    "No communication with Qbus Server, will try to reconnect every " + serverCheck + " minutes");
```
Setting the bridge offline already produces a log message, so you could remove your log message. You could also include the exception's message in the Thing status detail message. The hint, that it will reconnect every x minutes is not necessary as this should be obligatory.

`git rebase` is only necessary if there are changes in the upstream repo, that you want to have in your PR to work with.

QbusKoen commented:
@fwolter Thanks for your quick review! Updated the code ;-)

QbusKoen commented:
@fwolter Damn, forgot to add the error message to the off line Brdige. Pushed again ;-)

fwolter commented on bundles/org.openhab.binding.qbus/README.md:
```diff
@@ -1,10 +1,10 @@
 # Qbus Binding
 
-![Qbus Logo](doc/Logo.JPG)
+![Qbus Logo](doc/qbus.JPG)
```
You can remove the logo here entirely as you uploaded it to the docs repo. The website will put the logo here automatically.

QbusKoen replied:
ok, done. Took a look at other bindings and noticed there was no doc folder. So i removed mine as well. Hope this is ok?

QbusKoen commented:
Removed logo...

QbusKoen commented:
@fwolter ok, i was confused and acted to quickly. Removed the logo completely now.
Kinda confusing doh.

kaikreuzer commented:
@fwolter Can this be merged then? The website build is currently broken, which means we cannot publish the 3.0.2 release on our download page...

fwolter reviewed (approved):
Yes, LGTM.
